packages/deployer/subtasks/deploy-router.js

```diff
@@ -0,0 +1,15 @@
+const logger = require('../utils/logger');
+const { subtask } = require('hardhat/config');
+const { TASK_COMPILE } = require('hardhat/builtin-tasks/task-names');
+
+const { SUBTASK_DEPLOY_CONTRACTS, SUBTASK_DEPLOY_ROUTER } = require('../task-names');
+
+subtask(SUBTASK_DEPLOY_ROUTER).setAction(async (_, hre) => {
+  logger.subtitle('Deploying router');
+
+  await hre.run(TASK_COMPILE, { force: false, quiet: true });
```

Comment:
Shouldn't we compile ONLY the router here? (hh will check all the sources again to look for changes)

Reply:
That would be nice, but looks like the original task doesn't allow us to define which contracts to compile: https://github.com/nomiclabs/hardhat/blob/master/packages/hardhat-core/src/builtin-tasks/compile.ts#L1257-L1259

And FYI, I just did a `console.time` on this line and its already taking `700ms`, which I think is a lot keeping in mind that we only have 6 modules here.